lib/cursor.js

```diff
@@ -203,6 +204,16 @@ function Cursor(bson, ns, cmd, options, topology, topologyOptions) {
 // Inherit from Readable
 inherits(Cursor, Readable);
 
+Object.defineProperty(Cursor.prototype, 'optionsValidationLevel', {
+  enumerable: true,
+  get: function() {
+    if (this.s && this.s.options && this.s.options.optionsValidationLevel) {
+      return this.s.options.optionsValidationLevel;
+    }
+    return 'error';
```

Comment:
Not the level of the client?

Reply:
I'm not sure how we could inherit from the client. Similarly to `db`, `cursor` takes in `topology`, not `client`.

Reply:
Yeah, this is going to be a problem b/c we don't have an easy way of getting the client info to the cursor. But adding options management without the ability to configure it for a certain class will probably be frustrating to users.

Reply:
I could add `optionsValidationLevel` as an `overrideOnly` for any method that creates a cursor so the cursor will inherit that `optionsValidationLevel`. For example, `find` could have `optionsValidationLevel: { overrideOnly: true }` as part of its schema and then the override would be `optionsValidationLevel: this.optionsValidationLevel`. That would mean the `optionsValidationLevel` makes its way to the cursor as an option and will be set on `cursor.s.options`.

Reply:
@mbroadst and @daprahamian, do you think I should leave the `optionsValidationLevel` inheritance as is or try the approach above?

Reply:
My suspicion is that people will only ever really want to set this value once (e.g. through `MongoClient` constructor, or the connection string). What's perhaps worth discussion here is that the cursor can be created from a number of different "parents", who's value do we inherit here?

Adding the parents `optionsValidationLevel` to the options for creating this cursor might be good enough? Do we anticipate users will want to toggle options management at runtime?

Reply:
I don't think we should, at this time, allow users to toggle this at runtime. It should be set once, and that is it.

Reply:
I agree.

To address the parents question, I think it makes the most sense for a cursor to inherit the `optionsValidationLevel` of the parent from which it was created. A cursor can be created from a change stream, a collection, or a database. In the case of a cursor being created from a collection or a database, it will inherit that parent's `optionsValidationLevel'. For a cursor coming from a change stream, we will need to pass the `optionsValidationLevel` as an option to the change stream constructor and then pass it again to the cursor as an option.

Reply:
SGTM